src/Dhall/TypeCheck.hs

```diff
@@ -84,8 +85,21 @@ typeWithA
     -> Context (Expr s a)
     -> Expr s a
     -> Either (TypeError s a) (Expr s a)
-typeWithA tpa = loop
+typeWithA = typeWithAN (const Nothing)
+
+{-| Generalization of `typeWithA` that also allows custom normalization of types
+-}
+typeWithAN
```

Comment:
I'd suggest just making a breaking change to extend `typeWithA` to support a custom normalizer

Reply:
👍works for me